executables/bal.go

```diff
@@ -0,0 +1,306 @@
+package main
```

Comment:
Since the main function does a lot of work, shall we split it into small functions that do specific tasks. This will help with the readablility.